source/SavedGame.cpp

```diff
@@ -36,6 +36,8 @@ void SavedGame::Load(const string &path)
 	if(file.begin() != file.end())
 		this->path = path;
 	
+	int flagshipIterator = 1;
+	int flagshipTarget = 1;
```

Comment:
If the player has ships but no flagship (can happen if you sold your flagship and there are no other in-system ships, or all in-system ships are automata when you save) then this means the first ship is used as the flagship. Perhaps the flagshipTarget should be initialized as 0 or -1.

Reply:
Without a default value, any saves pre-patch will appear with no ship. Keeping that default so that pre-patch saves could (possibly incorrectly) show a ship would probably be better than nothing at all. Then we'll set `flagship -1` when writing out the save with no detected flagship. That seems like the solution that causes the least issues.